app/models/user.rb

```diff
@@ -197,14 +198,21 @@ def self.anonymous_console_admin
   # If the user is not in the DB then try to login the user on each available authentication source
   # If this succeeds then copy the user's details from the authentication source into the User table
   # Returns : User object OR nil
-  def self.try_to_login(login, password)
+  def self.try_to_login(login, password, api_request = false)
```

Comment:
would it make sense to add options hash instead of third argument? I'd like to avoid situation where we want to add 4th optional param in future